tests/Unit/Evolution/Initialization/Test_FunctionOfTimeFromFile.cpp

```diff
@@ -0,0 +1,140 @@
+  using component_list = tmpl::list<component<Metavariables>>;
+  enum class Phase { Initialization, Testing, Exit };
+};
+
+void test_options() noexcept {
+  CHECK(db::tag_name<Initialization::Tags::FuncOfTimeFile>() ==
+        "FuncOfTimeFile");
+  CHECK(db::tag_name<Initialization::Tags::FuncOfTimeSetNames>() ==
+        "FuncOfTimeSetNames");
+
+  const std::string option_string{
+      "FuncOfTimeFromFile:\n"
+      "  FuncOfTimeFile: TestFile.h5\n"
+      "  FuncOfTimeSetNames: [[Set1, Name1], [Set2, Name2]]"};
+  using option_tags =
+      tmpl::list<Initialization::OptionTags::FuncOfTimeFile,
+                 Initialization::OptionTags::FuncOfTimeSetNames>;
+  Options<option_tags> options{""};
+  options.parse(option_string);
+  CHECK(options.get<Initialization::OptionTags::FuncOfTimeFile>() ==
+        "TestFile.h5");
+  const auto& set_names =
+      options.get<Initialization::OptionTags::FuncOfTimeSetNames>();
+  const std::vector<std::array<std::string, 2>> expected_set_names{
+      {{{"Set1", "Name1"}}, {{"Set2", "Name2"}}}};
+  CHECK(set_names == expected_set_names);
+}
+}  // namespace
+
+SPECTRE_TEST_CASE("Unit.Evolution.FunctionOfTimeFromFile",
+                  "[Unit][Evolution][Actions]") {
+  test_options();
+
+  using MockRuntimeSystem = ActionTesting::MockRuntimeSystem<Metavariables>;
+
+  const ElementId<dim> self_id(1);
+
+  MockRuntimeSystem runner{{}};
+  ActionTesting::emplace_component_and_initialize<component<Metavariables>>(
+      &runner, self_id,
+      {std::string{
+           unit_test_path() +
+           "/Evolution/Initialization/FunctionOfTimeFromFile_TestData.h5"},
+       std::vector<std::array<std::string, 2>>{
+           {{{"ExpansionFactor", "ExpansionFactor"}},
+            {{"RotationAngle", "RotationAngle"}}}}});
+  runner.set_phase(Metavariables::Phase::Testing);
+  runner.next_action<component<Metavariables>>(self_id);
+
+  const auto& functions_of_time =
+      ActionTesting::get_databox_tag<component<Metavariables>,
+                                     domain::Tags::FunctionsOfTime>(runner,
+                                                                    self_id);
+
+  // Check that the FunctionOfTime and its derivatives have the expected
+  // values
+  const std::array<double, 2> expected_times{{0.0, 0.00599999974179903}};
+  const std::array<std::string, 2> expected_names{
+      {"RotationAngle", "ExpansionFactor"}};
+  const std::array<std::array<double, 3>, 2> expected_expansion{
+      {{{1.000000000000000, -0.000001426829676, 0.0}},
+       {{0.999999991439022, -0.000001426829676, 0.0}}}};
+  const std::array<std::array<double, 3>, 2> expected_rotation{
+      {{{0.0, 0.013472907726000, 0.0}},
+       {{0.000080837442877, 0.013472907726000, 0.0}}}};
+  std::unordered_map<std::string, std::array<std::array<double, 3>, 2>>
+      expected_functions;
+  expected_functions[expected_names[0]] = expected_rotation;
+  expected_functions[expected_names[1]] = expected_expansion;
+
```

Comment:
please check that the size of `functions_of_time` is the expected size, as your test will pass if the fetched map is empty...

Reply:
`REQUIRE(functions_of_time.size() == expected_names.size());`